components/ca-rating.js

```diff
@@ -0,0 +1,202 @@
+define([
+    './helpers/clear-element.js',
+    './helpers/create-element.js',
+    './helpers/cancel-event.js',
+    'document-register-element'
+], (clearElement, createElement, cancelEvent) => {
+    /**
+     * @exports ca-rating
+     * @description A custom HTML element (Web Component) that can be created using
+     * document.createElement('ca-rating') or included in a HTML page as an element.
+     */
+    class Rating extends HTMLElement {
+
+        /**
+         * @description Executes when the element is first created
+         * @return {undefined}
+         */
+        createdCallback() {
+            // render the component now that it has been created
+            this.render();
+        }
+
+        /**
+         * @description Executes when the element is attached to the DOM
+         * @access private
+         * @type {Event}
+         * @returns {undefined} nothing
+         */
+        attachedCallback() {
+            // render the component after it has been attached to the DOM
+            this.attached = true;
+            this.render();
+        }
+
+        /**
+         * @description Executes when any attribute is changed on the element
+         * @param {string} attrName - the name of the attribute to have changed
+         * @param {string} oldVal - the old value of the attribute
+         * @param {string} newVal - the new value of the attribute
+         * @returns {undefined}
+         */
+        attributeChangedCallback(attrName, oldVal, newVal) {
+            // only need to re-render the component if the attribute value has changed
+            if (oldVal !== newVal) {
+                this[attrName] = newVal;
```

Comment:
review if needed in stage 1 refactor. Might have been needed due to polyfill issues.